nmdc_automation/workflow_automation/models.py

```diff
@@ -39,11 +41,41 @@ def workflow_process_factory(record: Dict[str, Any]) -> PlannedProcess:
         "nmdc:ReadQcAnalysis": ReadQcAnalysis,
     }
     record.pop("_id", None)
+    # for backwards compatibility strip Activity from the end of the type
+    record["type"] = record["type"].replace("Activity", "")
+
+    # add type to Mags Analysis mags_list records
+    if record["type"] == "nmdc:MagsAnalysis" and "mags_list" in record:
+        for mag in record["mags_list"]:
+            if not mag.get("type"):
+                mag["type"] = "nmdc:MagBin"
+            if mag["gene_count"] == 'null':
+                mag["gene_count"] = 0
```

Comment:
exclude key if value is null instead of set to 0. Cheinchi updated the code so this value should be populated.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/262/commits/d05ff8c8b15cfdc5bc4235a1d27536dd0f9790fd

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/262/commits/1711fe3d4c6a834d34ee696140f535675203e96d